File: typescript/transpose/transpose.ts
export default class Transpose {
  static transpose(arr: string[]): string[] {
    const res: string[] = []
    arr.forEach((line, i) => {
      for (const [j, char] of [...line].entries()) {
        if (i === 0) {
          res.push(char)
        } else if (j >= res.length) {
          res.push(char.padStart(res[0].length, ' '))
        } else {
          res[j] += char
        }
      }
    });
    return res
  }
}

File: kotlin/transpose/src/main/kotlin/Transpose.kt
object Transpose {

    fun transpose(input: List<String>): List<String> {
        var res = mutableListOf<String>();
        input.forEachIndexed { i, row ->
            row.forEachIndexed { j, c ->
                if (res.size < j + 1) {
                    if (j != 0) {
                        res.add(Character.toString(c).padStart(res[j - 1].length))
                    } else {
                        res.add(Character.toString(c))
                    }
                } else {
                    res.set(j, res[j] + Character.toString(c).padStart(i + 1 - res[j].length))
                }
            }
        }
        return res
    }
}
